src/main/java/com/example/onlinebookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,100 @@
+package com.example.onlinebookstore.service.impl;
+
+import com.example.onlinebookstore.dto.OrderItemResponseDto;
+import com.example.onlinebookstore.dto.OrderRequestDto;
+import com.example.onlinebookstore.dto.OrderResponseDto;
+import com.example.onlinebookstore.dto.OrderStatusUpdateDto;
+import com.example.onlinebookstore.exception.EntityNotFoundException;
+import com.example.onlinebookstore.mapper.OrderItemMapper;
+import com.example.onlinebookstore.mapper.OrderMapper;
+import com.example.onlinebookstore.model.Order;
+import com.example.onlinebookstore.model.OrderItem;
+import com.example.onlinebookstore.model.ShoppingCart;
+import com.example.onlinebookstore.model.Status;
+import com.example.onlinebookstore.model.User;
+import com.example.onlinebookstore.repository.OrderItemRepository;
+import com.example.onlinebookstore.repository.OrderRepository;
+import com.example.onlinebookstore.service.OrderService;
+import com.example.onlinebookstore.service.ShoppingCartService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final ShoppingCartService shoppingCartService;
+
+    @Override
+    public OrderResponseDto placeOrder(User user, OrderRequestDto orderRequestDto) {
+        ShoppingCart cart = shoppingCartService.findByUserId(user);
+
+        Order order = new Order();
+        order.setUser(user);
+        order.setStatus(Status.PENDING); // Initial status is PENDING
+        order.setOrderDate(LocalDateTime.now());
+        order.setShippingAddress(orderRequestDto.getShippingAddress());
+
+        Set<OrderItem> orderItems = cart.getCartItems().stream()
+                .map(cartItem -> {
+                    OrderItem orderItem = new OrderItem();
+                    orderItem.setOrder(order);
+                    orderItem.setBook(cartItem.getBook());
+                    orderItem.setQuantity(cartItem.getQuantity());
+                    orderItem.setPrice(cartItem.getBook().getPrice()
+                            .multiply(BigDecimal.valueOf(cartItem.getQuantity())));
+                    return orderItem;
+                })
+                .collect(Collectors.toSet());
+        order.setOrderItems(orderItems);
+
+        BigDecimal total = orderItems.stream()
+                .map(OrderItem::getPrice)
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
+        order.setTotal(total);
```

Comment:
and move this to some `calculateTotalPrice(orderItems)` method as well

Reply:
done